runtime/src/bank.rs

```diff
@@ -897,77 +897,116 @@ impl Bank {
             let inflation = self.inflation.read().unwrap();
 
             (*inflation).validator(year) * self.capitalization() as f64 * period
-        };
+        } as u64;
+
+        let vote_balance_and_staked = self.stakes.read().unwrap().vote_balance_and_staked();
```

Comment:
MUST FIX: use of `vote_balance_and_staked()` here incorrectly accounts for the warm-up handling of stakes.

we need like this:

```patch
diff --git a/runtime/src/stakes.rs b/runtime/src/stakes.rs
index cb68d79ce4..9f2997632c 100644
--- a/runtime/src/stakes.rs
+++ b/runtime/src/stakes.rs
@@ -94,10 +94,22 @@ impl Stakes {
             .sum()
     }
 
+    pub fn total_vote_balance_and_stakes(&self) -> u64 {
+        self.stake_delegations
+            .iter()
+            .map(|(_, stake_delegation)| {
+                stake_delegation.stake
+            })
+            .sum::<u64>() +
+        self.vote_accounts
+            .iter()
+            .map(|(_pubkey, (_staked, vote_account))| vote_account.lamports)
+            .sum::<u64>()
+    }
+

```

Reply:
Fixed https://github.com/solana-labs/solana/pull/10914/commits/0218d32705392151c5e8e4e043d7709aefbcef72